source/documentation/using_the_api/register_components.md

```diff
@@ -1,18 +1,34 @@
-## The relationship between entries, items and records
+## The components of a register
 
-Registers are made up of:
+Registers are primarily composed of:
 
 * entries
 * items
 * records
 
+### Entries
+
 Entries contain metadata about what and when data has changed. 
 
-Each entry in a register has a relationship with one item, which contains data. An item is connected to an entry with an item hash.   
+Entries contain the fields `entry-number`,  `entry-timestamp`, `item-hash` and `key`. 
+
+The `entry-number` is unique and defines an entry’s position within the ordered list of a register. The `entry-timestamp` is the time when an entry was introduced to a given register, but it is no guarantee of the order of entries in a register. The `entry-number` is.
+
+Entries are connected to items by a given `item-hash`. 
+
+Each `key` is an alphanumeric string unique to the data in a register. The primary key (i.e. the unique identifier) always has the same name as the name of the register. For example, every entry in the `local-authority-eng` register has a field called `local-authority-eng`. Foreign keys, which are not always present, link to other registers. 
+
+> Entries also contain the field `index-entry-number`, which is a unique number that defines an entry's position within the ordered list of an index. Indexes are not a reliable feature yet, which are not ready for use by most users.  
+
+### Items
+
+Items contain structured data for a given primary key in a register. Items have predefined fields which are consistent within a register.
```

Comment:
Did you decide to not mention identifiers?

Reply:
@arnau 

> Did you decide to not mention identifiers?

I tried to mention it earlier in the text, in the context of entries:

> Each `key` is an alphanumeric string unique to the data in a register. The primary key (i.e. the unique identifier) always has the same name as the name of the register. For example, every entry in the `local-authority-eng` register has a field called `local-authority-eng`. Foreign keys, which are not always present, link to other registers. 

> ... alphanumeric string _unique_ to the data ...
> ... (i.e. the unique identifier) ...

We could say instead(?):

> ... The primary key is a unique identifier for the register, and always has the same name as the name of the register ...

I think we need to be specific that this would be the primary key here? This old definition was floating around and I don't think this is accurate:

> Records define themselves by a unique identifier associated with a single record in a register. A unique identifier can be a candidate for a primary key. The primary key always has the same name as the name of the register. For example, every entry in the country register has a field called country. This helps to link that entry specifically to the country register.

Reply:
Same here. Some of my understanding about primary keys was drawn from that, but talking to you the other day I realise that paragraph is quite problematic.